src/hooks/useCalculation.ts

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from 'react';
+import type { UnitBoxes } from '../types';
+import { useStateValue } from '../state/state';
+import { ARDataPriceEstimator } from '../utils';
+
+/** ARDataPriceEstimator instance, will fire off initial fetch calls when constructed */
+const arDataPriceEstimator = new ARDataPriceEstimator();
+
+const winstonToArValue = 0.000_000_000_001;
+
+interface previousValues {
+	bytes: number;
+	fiat: number;
+	ar: number;
+}
+
+/**
+ * Use calculation hook listens to changes to the global unitBox state. If a unitBox
+ * changes, it will re-calculate the boxes calculate the other boxes. Then, update
+ * the unitBox state with one dispatch call
+ *
+ * The hook is intended to be used once in the Calculator component
+ */
+export default function useCalculation(): void {
+	const [{ unitBoxes }, dispatch] = useStateValue();
+	const [sendingCalculation, setSendingCalculation] = useState(false);
+
+	// Save previous  values for determining if a value has changed
+	const [prevUnitVals, setPrevUnitVals] = useState<previousValues>({
+		bytes: unitBoxes.ar.value,
+		fiat: unitBoxes.fiat.value,
+		ar: unitBoxes.ar.value
+	});
+
+	/** @TODO Conversion will come from fiatToArData[unitBoxes.fiat.currUnit] */
+	const currentFiatToWinstonConversion = 0.000_000_000_15;
+
+	/** Whenever unitBoxes change, starts a new calculation */
+	useEffect(() => {
+		calculateUnitBoxes();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [unitBoxes]);
+
+	async function calculateUnitBoxes(): Promise<void> {
+		if (sendingCalculation) {
+			// Don't calculate if calculation is already dispatching, or cooling down
+			return;
+		}
+		setSendingCalculation(true);
+
+		try {
+			// Price estimator has several error throwing cases, wrap all calls in try/catch
+
+			/** @TODO Add ArDrive Community fee to determined winstonPrice */
+			const [byteCount, winstonPrice] = await determineWinstonAndBytes();
+
+			const newUnitBoxes: UnitBoxes = {
+				bytes: { ...unitBoxes.bytes, value: byteCount },
+				ar: { ...unitBoxes.ar, value: Number((winstonPrice * 0.000_000_000_001).toFixed(12)) },
+				fiat: { ...unitBoxes.fiat, value: Number((winstonPrice * currentFiatToWinstonConversion).toFixed(6)) }
```

Comment:
What's behind the choice of toFixed(6)?

Reply:
When using the current USD -> AR conversion, 6 decimal places shows a value for 1 byte: `1 byte = 0.000_002 USD`

I expected this one to be set temporary

Reply:
We're going to have to solve a few UI and arithmetic issues when handling really large and really small numbers. Let's deal with those in a followup ticket.